AE->OSO ansible merge: ansible in container

aveshagarwal opened the pull request:
Changes to Dockerfile to make it work with openshift-ansible and also added privileged flag to allow sharing host network and host dirs.

aveshagarwal commented:
@detiber @sdodson

detiber commented:
@aveshagarwal is there an associated README item for this? since it makes the assumption of having a pre-generated /etc/ansible/hosts file, we'd need to let the user know that they will need that in place before running.

aveshagarwal commented:
There is no README for this. And you are right, because it is not possible to pre ship /etc/ansible/hosts so it should be user supplied.

aveshagarwal commented:
@detiber I have added  README_ANSIBLE that contains instructions to build and run ansible in a docker container, and also modified Dockerfile to replace CMD with ENTRYPOINT to run the container as executable. Also moved Dockerfile to a new dir ansible-container to make it easier to build the container.

detiber commented:
@aveshagarwal may be better to call it README_DOCKER.md or README_ANSIBLE_CONTAINER.md

aveshagarwal commented:
@detiber Initially I was planning to name it exactly README_ANSIBLE_CONTAINER.md but did not thinking that it might be too long. But I will make this change now and update the PR.

aveshagarwal commented:
@detiber updated the PR.

detiber commented:
:+1: @brenton ready to merge as far as I'm concerned.

brenton commented:
@aweiteka, I see this is based on one of your commits.  With these modifications is this helpful to you?  I know you've been working on containerized installation tools and I just want to double check this is based on your latest work.

aweiteka commented on ansible-container/Dockerfile:
```diff
@@ -0,0 +1,15 @@
+# only git and ansible are known requirements
+RUN yum install -y --enablerepo rhel-7-server-extras-rpms net-tools bind-utils tmux git ansible
+
+RUN git clone https://github.com/openshift/openshift-ansible.git /opt/openshift-ansible
```
It would be better to use the `ADD` directive here.

1. No need to clone when it's local.
1. We want to control the packaging, not just pull from master.

```
ADD / /opt/openshift-ansible/
```

aveshagarwal replied:
@aweiteka I am trying to understand this. It seems to me that the above line adds everything inside / on the host into /opt/openshift-ansible/ in the building container. Is that desired?

ashcrow replied:
@aveshagarwal I read it that way too.

Maybe it should be 

```
ADD ./openshift-ansible/ /opt/openshift-ansible/
```

or something like that. Does that work @aweiteka?

aveshagarwal replied:
@aschrow I have discussed it with Aaron already and now testing the container.

ashcrow replied:
@aveshagarwal great!

aweiteka commented on ansible-container/Dockerfile:
```diff
@@ -0,0 +1,15 @@
+
+RUN git clone https://github.com/openshift/openshift-ansible.git /opt/openshift-ansible
+
+ENTRYPOINT ["/usr/bin/ansible-playbook", "/opt/openshift-ansible/playbooks/byo/config.yml"]
```
This constrains the user to a specific playbook. We can couple `ENTRYPOINT` with `CMD` for more flexibility.

```
ENTRYPOINT ["/usr/bin/ansible-playbook"]
OPT /opt/openshift-ansible/playbooks/byo/config.yml
```

CMD defines the default playbook. The user can override this by appending another playbook.

```
docker run ansible some/other/playbook.yml
```

aveshagarwal replied:
@aweiteka sounds good to me.

aveshagarwal commented:
@aweiteka @brenton I have updated Dockefile and README_ANSIBLE_CONTAINER as per the feedback and also tested it with latest openshift-ansible repo and openshift-3.0.1.0-1.git.525.eddc479.el7ose.x86_64 .

ashcrow commented:
@aweiteka @brenton  I believe @aveshagarwal has taken care of the outstanding requests. Can you review?

brenton commented:
Looks like the changes reflect @aweiteka's guidance.  merging...
